### PR TITLE
[proxy] Add new port

### PR DESCRIPTION
**Pull request by tian-lt**
**Describe the pull request**

- #### What does your PR fix?
  No, this PR is going to add a new port - proxy, which is referring to a GitHub repo: https://github.com/microsoft/proxy

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  all, No.
 *(This is a header-only library*

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

Cc @mingxwa

### CODE REVIEW

File: ports/proxy/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "proxy",
+  "version": "1.0.1",
+  "description": "Proxy: Easy Polymorphism in C++",
```
Comment (mingxwa) [inline ports/proxy/vcpkg.json:4]: Suggested wording: "A single-header C++20 library that facilitates runtime polymorphism."

**Review by mingxwa — DISMISSED** [review]

File: ports/proxy/vcpkg.json (hunk 1 above)
Comment (tian-lt) [reply ports/proxy/vcpkg.json]: done.

**Review by mingxwa — DISMISSED** [review]

**Comment by Adela0814** [comment]
@tian-lt  Thanks for your PR! feel free to ping us if the PR is ready for review.

**Comment by tian-lt** [comment]
@Adela0814 thank you for reminding.
Marked this PR as ready for review.

**Comment by mingxwa** [comment]
@Adela0814 I saw the tag "requires:author-response" was added. Which comment are you referring to? Is there any other information required from our side?

cc @tian-lt @BillyONeal

**Review by Adela0814 — CHANGES_REQUESTED** [review]

File: ports/proxy/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,19 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO microsoft/proxy
+    REF release/1.0.1
+    SHA512 5822064b373d3da8cea884d0c727e98d988252a7324ea9b78fa63fc86ee6afd73852a49f3685842345b86d625380140f80998263a6c8150fe66a30d6ba7e928a
+    HEAD_REF main
+)
+
+vcpkg_cmake_configure(
+    SOURCE_PATH ${SOURCE_PATH}
```
[the remark this answers is not in the document]
Comment (tian-lt) [reply ports/proxy/portfile.cmake]: Thanks for the suggestion.
Added the quotes.

**Review by Adela0814 — APPROVED** [review]

**Pull request merged**